integration/release_upgrade_utils_linux_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"compress/gzip"
+	"context"
+	"fmt"
+	"net/http"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/mod/semver"
+	exec "golang.org/x/sys/execabs"
+
+	"github.com/containerd/containerd/archive"
+	"github.com/containerd/containerd/version"
+)
+
+// downloadPreviousReleaseBinary downloads the latest version of previous release
+// into the target dir.
+func downloadPreviousReleaseBinary(t *testing.T, targetDir string) {
```

Comment:
We may eventually want a way to supply a local containerd archive instead of downloading one, especially if we want this test to be usable for distros who build their own containerd.  But I don't think we need to do that now.